src/event_handler.rs

```diff
@@ -464,93 +470,127 @@ async fn resend_pending_transaction(
     tx_sender: Sender<JobsTxnMetadata>,
 ) {
     loop {
-        // Get the first pending txn from the queue (with the least nonce) and focus on unblocking this nonce
-        let Some(mut pending_txn_data) = pending_txns.lock().unwrap().pop_front() else {
+        // Get the currently injected gas address in the executor
+        let current_gas_address = app_state
+            .http_rpc_client
+            .lock()
+            .unwrap()
+            .clone()
+            .unwrap()
+            .address();
+        let mut pending_txn_data = pending_txns.lock().unwrap().pop_front();
+
+        // Pop and send the pending txns with old gas account to the main 'Jobs' txn sender for resending with consistent nonce
+        while pending_txn_data.is_some() {
+            if pending_txn_data.clone().unwrap().http_rpc_client.address() == current_gas_address {
+                break;
+            }
+
+            let _ = tx_sender.send(pending_txn_data.unwrap().txn_data).await;
+            pending_txn_data = pending_txns.lock().unwrap().pop_front();
+        }
+
+        // Continue if the pending txns deque is empty
+        if pending_txn_data.is_none() {
             sleep(Duration::from_millis(200)).await;
             continue;
         };
 
-        let mut resend_txn = true;
+        let mut pending_txn_data = pending_txn_data.unwrap();
+
+        // Wait for some time for the pending txn to be mined before resending
+        sleep(Duration::from_secs(
+            RESEND_TXN_INTERVAL
+                - min(
+                    RESEND_TXN_INTERVAL,
+                    pending_txn_data.last_monitor_instant.elapsed().as_secs(),
+                ),
+        ))
+        .await;
+
+        // Get the transaction receipt for the pending transaction to check if it's still pending or been dropped
+        let Ok(txn_receipt) = pending_txn_data
+            .http_rpc_client
+            .get_transaction_receipt(pending_txn_data.txn_hash)
+            .await
+        else {
+            // Continue if failed to retrieve transaction receipt from the rpc
+            pending_txn_data.last_monitor_instant = Instant::now();
+            pending_txns.lock().unwrap().push_front(pending_txn_data);
+            continue;
```

Comment:
can it be infinite loop, in some case? Like txn isn't even mined yet. Earlier there was retry until retry_deadline, but it is removed now?

Reply:
Oh Yes Sorry, will add back that check. Thanks for pointing out!